astropy/io/ascii/ui.py

```diff
@@ -608,8 +608,7 @@ def _get_guess_kwargs_list(read_kwargs):
 
     # Start with ECSV because an ECSV file will be read by Basic.  This format
     # has very specific header requirements and fails out quickly.
-    if HAS_YAML:
-        guess_kwargs_list.append(dict(Reader=ecsv.Ecsv))
+    guess_kwargs_list.append(dict(Reader=ecsv.Ecsv))
```

Comment:
In order to handle this, we need to always try ECSV even if PyYAML is not installed.  Guessing just got a little bit slower (but not too bad).